### DISCUSSION TITLE
Running the prometheus-nginx-exporter service on CentOS 7 [VMBitrix 7.5.2]

### DISCUSSION THREAD

**Discussion by ghost** [nginx/nginx-prometheus-exporter#508] [Q&A] [answered]
Please help me solve the problem of starting the service for the prometheus-nginx-exporter exporter module

Installed prometheus-nginx-exporter package via Snap

In the NGINX configs I added

**/etc/nginx/bx/site_settings/default/prometheus-nginx-exporte.conf**
```
location /stub_status {
stub_status;
allow 127.0.0.1;
deny all;
}
```

**/etc/nginx/bx/site_avaliable/s1.conf**
`listen 8080 default_server;`

The metric stub is sent over port 8080, everything is ok, I checked it through the browser and Curl http://127.0.0.1:8080/stub_status

But the problem is the following: since prometheus-nginx-exporter does not have a daemon, I had to create Units, I created them

**/etc/systemd/system/nginx_exporter.service**
```
Description=NGINX Prometheus Exporter
Requires=nginx_exporter.socket

[Service]
User=nginx_exporter
ExecStart=/var/lib/snapd/snap/nginx-prometheus-exporter/70/nginx-prometheus-exporter --nginx.scrape-uri="http://127.0.0.1:8080/stub_status"

[Install]
WantedBy=multi-user.target
```
and

**/etc/systemd/system/nginx_exporter.socket**
```
Description=NGINX Prometheus Exporter

[Socket]
ListenStream=9113

[Install]
WantedBy=sockets.target
```
But I can’t start the service, it crashes with an error, can you tell me the reasons for it?

**$ systemctl daemon-reload**
**$ systemctl start nginx_exporter**
**$ systemctl status nginx_exporter**

Errors in the system log when trying to start the service are:

```
24 21:54:04 admin87.fvds.ru polkitd[481]: Registered Authentication Agent for unix-process:18615:236960770 (system bus name :1.758613 [/usr/bin/pkttyagent --notify-fd 5 --fallback], object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale ru_RU.UTF-8)
24 21:54:04 admin87.fvds.ru systemd[1]: Started nginx_exporter.service.
24 21:54:04 admin87.fvds.ru systemd[18621]: Failed at step USER spawning /var/lib/snapd/snap/nginx-prometheus-exporter/70/nginx-prometheus-exporter: No such process
24 21:54:04 admin87.fvds.ru systemd[1]: nginx_exporter.service: main process exited, code=exited, status=217/USER
24 21:54:04 admin87.fvds.ru systemd[1]: Unit nginx_exporter.service entered failed state.
24 21:54:04 admin87.fvds.ru systemd[1]: nginx_exporter.service failed.
24 21:54:04 admin87.fvds.ru polkitd[481]: Unregistered Authentication Agent for unix-process:18615:236960770 (system bus name :1.758613, object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale ru_RU.UTF-8) (disconnected from bus)
```

The main thing there, it seems to me, is this, but I can’t figure out how to fix it...

**Failed at step USER spawning /var/lib/snapd/snap/nginx-prometheus-exporter/70/nginx-prometheus-exporter: No such process**

I tried changing to (based on advice from here https://superuser.com/questions/1156676/what-causes-systemd-failed-at-step-user-spawning-usr-sbin-opendkim-no-such-p)

```
User=root
Group=root
```

or even remove the User and Group directives completely
did not help...

I need help!

**Comment by lucacome** [comment]
Hi @toorrp4 

I think this is a problem with snap and how they manage the process. I'll try to reproduce and find a workaround and let you know.

**Comment by ghost** [comment]
Thank you, I'm really looking forward to it! It would be useful to add to your instructions. By the way, another exporter, node-exporter, follows this [instruction](https://www.theairtips.com/post/how-to-install-node-exporter-on-centos-7-linux-monitoring) on the same server with CentOS 7, everything was installed and working, I just had to add a rule to iptables for the port 
`$ iptables -I INPUT -p tcp -m tcp --dport 9100 -j ACCEPT` 
(maybe this will also be useful to find out the reasons).

**Reply by lucacome** [reply]
Did you manually download the binary for `node_exporter`? I think that if you download the binary from our release page and follow the same it should work too. I think `snap` is getting in the way in this case... (haven't confirmed yet)

**Reply by ghost** [reply]
yes I installed it via Snap according to their [instructions](https://snapcraft.io/install/nginx-prometheus-exporter/centos)

**Comment by lucacome** [comment, marked as answer]
Looks like the account that opened this was deleted, but for reference I was able to use the exporter installed via Snap in systemd with this config for **/etc/systemd/system/nginx_exporter.service**

```
Description=NGINX Prometheus Exporter
Requires=nginx_exporter.socket

[Service]
ExecStart=/snap/nginx-prometheus-exporter/current/nginx-prometheus-exporter --nginx.scrape-uri=http://127.0.0.1:8080/stub_status

[Install]
WantedBy=multi-user.target
```

and the rest of the config as in the opening post.
